osquery/tables/system/darwin/smc_keys.cpp

```diff
@@ -315,5 +315,145 @@ QueryData genSMCKeys(QueryContext &context) {
+  keyToName["TCSA"] = "PECI SA";
+  keyToName["TCGC"] = "PECI GPU";
+  keyToName["TCGc"] = "PECI GPU";
+  keyToName["TG0P"] = "GPU Proximity";
+  keyToName["TG0D"] = "GPU Die";
+  keyToName["TG1D"] = "GPU Die";
+  keyToName["TG0H"] = "GPU Heatsink";
+  keyToName["TG1H"] = "GPU Heatsink";
+  keyToName["Ts0S"] = "Memory Proximity";
+  keyToName["TM0P"] = "Mem Bank A1";
+  keyToName["TM1P"] = "Mem Bank A2";
+  keyToName["TM8P"] = "Mem Bank B1";
+  keyToName["TM9P"] = "Mem Bank B2";
+  keyToName["TM0S"] = "Mem Module A1";
+  keyToName["TM1S"] = "Mem Module A2";
+  keyToName["TM8S"] = "Mem Module B1";
+  keyToName["TM9S"] = "Mem Module B2";
+  keyToName["TN0D"] = "Northbridge Die";
+  keyToName["TN0P"] = "Northbridge Proximity 1";
+  keyToName["TN1P"] = "Northbridge Proximity 2";
+  keyToName["TN0C"] = "MCH Die";
+  keyToName["TN0H"] = "MCH Heatsink";
+  keyToName["TP0D"] = "PCH Die";
+  keyToName["TPCD"] = "PCH Die";
+  keyToName["TP0P"] = "PCH Proximity";
+  keyToName["TA0P"] = "Airflow 1";
+  keyToName["TA1P"] = "Airflow 2";
+  keyToName["Th0H"] = "Heatpipe 1";
+  keyToName["Th1H"] = "Heatpipe 2";
+  keyToName["Th2H"] = "Heatpipe 3";
+  keyToName["Tm0P"] = "Mainboard Proximity";
+  keyToName["Tp0P"] = "Powerboard Proximity";
+  keyToName["Ts0P"] = "Palm Rest";
+  keyToName["Tb0P"] = "BLC Proximity";
+  keyToName["TL0P"] = "LCD Proximity";
+  keyToName["TW0P"] = "Airport Proximity";
+  keyToName["TH0P"] = "HDD Bay 1";
+  keyToName["TH1P"] = "HDD Bay 2";
+  keyToName["TH2P"] = "HDD Bay 3";
+  keyToName["TH3P"] = "HDD Bay 4";
+  keyToName["TO0P"] = "Optical Drive";
+  keyToName["TB0T"] = "Battery TS_MAX";
+  keyToName["TB1T"] = "Battery 1";
+  keyToName["TB2T"] = "Battery 2";
+  keyToName["TB3T"] = "Battery";
+  keyToName["Tp0P"] = "Power Supply 1";
+  keyToName["Tp0C"] = "Power Supply 1 Alt.";
+  keyToName["Tp1P"] = "Power Supply 2";
+  keyToName["Tp1C"] = "Power Supply 2 Alt.";
+  keyToName["Tp2P"] = "Power Supply 3";
+  keyToName["Tp3P"] = "Power Supply 4";
+  keyToName["Tp4P"] = "Power Supply 5";
+  keyToName["Tp5P"] = "Power Supply 6";
+  keyToName["TS0C"] = "Expansion Slots";
+  keyToName["TA0S"] = "PCI Slot 1 Pos 1";
+  keyToName["TA1S"] = "PCI Slot 1 Pos 2";
+  keyToName["TA2S"] = "PCI Slot 2 Pos 1";
+  keyToName["TA3S"] = "PCI Slot 2 Pos 2";
+
+  std::map<std::string, std::string>::const_iterator it = keyToName.begin();
+  for (; it != keyToName.end(); ++it) {
+    genSMCKey(it->first, smc, results);
+
+    Row smcRow = results.back();
+    if (smcRow["value"].empty() /*|| smcRow["type"].compare("sp78") != 0*/) {
+      results.pop_back();
+      continue;
+    }
+
+    Row r;
+    r["smc_key"] = smcRow["key"];
+    r["name"] = keyToName[smcRow["key"]];
+
+    // Convert hex string to decimal.
+    const char *hexstring = smcRow["value"].c_str(), *pos = hexstring;
+    unsigned char val[12];
+    size_t count = 0;
+
+    for(count = 0; count < sizeof(val) / sizeof(val[0]); count++) {
+        sscanf(pos, "%2hhx", &val[count]);
```

Comment:
See if you can accomplish this with boost's `unhex`: http://www.boost.org/doc/libs/1_51_0/libs/algorithm/doc/html/the_boost_algorithm_library/Misc/hex.html